test/veritaserum_test.exs

```diff
@@ -29,6 +29,21 @@ defmodule VeritaserumTest do
     end
   end
 
+  describe "when return option is :score_and_marks" do
```

Comment:
As far as I understand, `describe` blocks are not intended to use to represent different contexts. This should be just inside the `analyze/2` describe block after splitting the function in to like proposed above.